scripts/invariant_checks.py

```diff
@@ -81,30 +89,45 @@ def main(argv: Sequence[str] | None = None) -> None:
             deployed_pools = Hyperdrive.get_hyperdrive_pools_from_registry(chain, registry_address)
             last_pool_check_block_number = latest_block_number
 
+        # Ensure we check on every block, and wait for a new mined block
+        # Throw a critical if it gets behind
         if not latest_block_number > last_executed_block_number:
             # take a nap
-            time.sleep(parsed_args.invariance_check_sleep_time)
+            time.sleep(1)
             continue
 
         # Update block number
         last_executed_block_number = latest_block_number
         # Loop through all deployed pools and run invariant checks
-        for hyperdrive_obj in deployed_pools:
-            name = hyperdrive_obj.name
-            logging.info("Running invariance check on %s", name)
-            run_invariant_checks(
-                latest_block=latest_block,
-                interface=hyperdrive_obj.interface,
-                raise_error_on_failure=False,
-                log_to_rollbar=log_to_rollbar,
-                pool_name=name,
-            )
+
+        for check_block in range(start_block_number, latest_block_number + 1):
+            if (latest_block_number - start_block_number) > LOOKBACK_BLOCK_LIMIT:
+                error_message = "Unable to keep up with invariant checks."
+                logging.error(error_message)
+                log_rollbar_message(error_message, logging.ERROR)
+                break
+
+            check_block_data = chain.block_data(block_identifier=check_block)
+            partials = [
+                partial(
+                    run_invariant_checks,
+                    check_block_data=check_block_data,
+                    interface=hyperdrive_obj.interface,
+                    log_to_rollbar=log_to_rollbar,
+                    pool_name=hyperdrive_obj.name,
+                )
+                for hyperdrive_obj in deployed_pools
+            ]
+
+            logging.info("Running invariant checks for block %s on pools %s", check_block, deployed_pools)
+            asyncio.run(async_runner(return_exceptions=False, funcs=partials))
+
+        start_block_number = latest_block_number + 1
```

Comment:
can you rename this? like uhh... `last_checked_block` or something? I think the `if (latest_block_number - start_block_number) > LOOKBACK_BLOCK_LIMIT:` conditional makes it seem like this only runs for `LOOKBACK_BLOCK_LIMIT` iterations.